From 30dbfc2770cb49cb3babfb750205b126d7d88791 Mon Sep 17 00:00:00 2001
From: Morten Linderud <morten@linderud.pw>
Date: Thu, 20 Feb 2020 09:44:29 +0100
Subject: [PATCH] [functions] Ignore modinfo built-in modules

modinfo have started reporting filenames as "(builtin)" on built-in
modules. Ignore these and continue with other modules.

Fixes: https://bugs.archlinux.org/task/65564
Introduced in: https://git.kernel.org/pub/scm/utils/kernel/kmod/kmod.git/commit/?id=e7e2cb61fa9f1db3429d91ef6accff549500d268

Signed-off-by: Morten Linderud <morten@linderud.pw>
---
 functions | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/functions b/functions
index f407c82..1c72b56 100644
--- a/functions
+++ b/functions
@@ -371,7 +371,7 @@ add_module() {
     target=${1%.ko*} target=${target//-/_}
 
     # skip expensive stuff if this module has already been added
-    (( _addedmodules["$target"] == 1 )) && return
+    (( _addedmodules["$target"] > 0 )) && return
 
     while IFS=':= ' read -r -d '' field value; do
         case "$field" in
